lib/rules/valid-component-name.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @author Wayne Zhang
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const utils = require('../utils')
+const { toRegExp } = require('../utils/regexp')
+
+const htmlElements = require('../utils/html-elements.json')
+const deprecatedHtmlElements = require('../utils/deprecated-html-elements.json')
+const svgElements = require('../utils/svg-elements.json')
+
+const RESERVED_NAMES_IN_VUE = new Set(
+  require('../utils/vue2-builtin-components')
+)
+const RESERVED_NAMES_IN_VUE3 = new Set(
+  require('../utils/vue3-builtin-components')
+)
+const kebabCaseElements = [
+  'annotation-xml',
+  'color-profile',
+  'font-face',
+  'font-face-src',
+  'font-face-uri',
+  'font-face-format',
+  'font-face-name',
+  'missing-glyph'
+]
```

Comment:
Are those actually tag names that should always be allowed? Where does this list come from?

Reply:
I referred to a [previous PR comment](https://github.com/vuejs/eslint-plugin-vue/pull/757#discussion_r245944977). It's listed in the [Custom Elements Standard](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name).

I'll take another look to see if there's a more detailed explanation. If there are related inputs, feel free to add them.

Reply:
Oh, I misunderstood this at first. These reserved names can be used in HTML, but they cannot be used as names for custom elements.
I referred to a [previous PR comment](https://github.com/vuejs/eslint-plugin-vue/pull/757#discussion_r245944977). It's listed in the [Custom Elements Standard](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name).

Deleted ✅

Reply:
Ah, that makes sense :+1: